eth_tester/backends/mock/serializers.py

```diff
@@ -42,11 +42,14 @@ def serialize_full_transaction(transaction, block, transaction_index, is_pending
     if 'gas_price' in transaction:
         return serialized_transaction
     else:
-        return assoc(
-            serialized_transaction,
-            'gas_price',
-            calculate_effective_gas_price(transaction, block)
+        # TODO: Sometime in 2022 the inclusion of gas_price may be removed from dynamic fee
+        #  transactions and we can get rid of this behavior.
+        #  https://github.com/ethereum/execution-specs/pull/251
+        gas_price = (
```

Comment:
This was a whoopsie before that I caught when testing via web3.py. If the transaction is pending, `gas_price` should be equal to `max_fee_per_gas` for dynamic fee transactions, otherwise (if already mined) it is the effective gas price. The explanation is in the link in the TODO.